commands/src/main/java/org/wildfly/extras/creaper/commands/ra/AddConnectionFactoryToRA.java

```diff
@@ -0,0 +1,142 @@
+package org.wildfly.extras.creaper.commands.ra;
+
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+/**
+ * Adds new connection factory to existing resource adapter
+ */
+public final class AddConnectionFactoryToRA implements OnlineCommand {
+    private final String poolName;
+    private final String localJndiName;
+    private final String resourceAdapterId;
+    private final String userName;
+    private final String password;
+    private final Map<String, String> properties;
+    private final boolean security;
+    private final boolean tracking;
+
+    private AddConnectionFactoryToRA(Builder builder) {
+        this.poolName = builder.poolName;
+        this.localJndiName = builder.localJndiName;
+        this.resourceAdapterId = builder.resourceAdapterId;
+        this.userName = builder.userName;
+        this.password = builder.password;
+        this.security = builder.security;
+        this.properties = builder.properties;
+        this.tracking = builder.tracking;
+    }
+
+    @Override
+    public void apply(OnlineCommandContext ctx) throws Exception {
+        Operations ops = new Operations(ctx.client);
+        Address address = Address.subsystem("resource-adapters").and("resource-adapter", resourceAdapterId)
+                .and("connection-definitions", poolName);
+        ops.add(address, Values.of("class-name", "org.jboss.resource.adapter.jms.JmsManagedConnectionFactory")
+                .and("jndi-name", localJndiName));
+        if (security) {
+            ops.writeAttribute(address, "security-application", true);
+        }
+        if (userName != null) {
+            ops.writeAttribute(address, "recovery-username", userName);
+        }
+        if (password != null) {
+            ops.writeAttribute(address, "recovery-password", password);
+        }
+        ops.writeAttribute(address, "tracking", tracking);
+        for (Map.Entry<String, String> pair : properties.entrySet()) {
+            Address configPropsAddress = Address.subsystem("resource-adapters")
+                    .and("resource-adapter", resourceAdapterId)
+                    .and("connection-definitions", poolName).and("config-properties", pair.getKey());
+            ops.add(configPropsAddress, Values.of("value", pair.getValue()));
+        }
```

Comment:
Did you consider creating a `Batch` and then performing all ops in a single "transaction"? It's not strictly required, but very nice to have :-)